xbmc/video/guilib/VideoStreamSelectHelper.cpp

```diff
@@ -0,0 +1,423 @@
+{
+  CGUIDialogSelect* dialog{CServiceBroker::GetGUI()->GetWindowManager().GetWindow<CGUIDialogSelect>(
+      WINDOW_DIALOG_SELECT_VIDEO_STREAM)};
+  if (!dialog)
+  {
+    CLog::LogF(LOGERROR, "Unable to get WINDOW_DIALOG_SELECT_VIDEO_STREAM dialog instance");
+    return;
+  }
+
+  auto& components = CServiceBroker::GetAppComponents();
+  auto appPlayer = components.GetComponent<CApplicationPlayer>();
+  const int streamCount = appPlayer->GetVideoStreamCount();
+  const int selectedIndex = appPlayer->GetVideoStream();
+
+  std::vector<std::pair<int, VideoStreamInfo>> streams;
+  // Collect all streams
+  for (int i = 0; i < streamCount; ++i)
+  {
+    VideoStreamInfo info;
+    appPlayer->GetVideoStreamInfo(i, info);
+    streams.emplace_back(i, info);
+  }
+
+  // Sort streams
+  std::sort(streams.begin(), streams.end(), SortComparerStreamVideo());
+
+  // Convert streams to FileItem's
+  CFileItemList itemsToDisplay;
+
+  for (auto& [index, info] : streams)
+  {
+    CFileItemPtr fileItem = std::make_shared<CFileItem>(info.name);
+    fileItem->SetProperty("stream.index", index);
+    fileItem->SetProperty("stream.description", info.name);
+    fileItem->SetProperty("stream.codec", info.codecName);
+
+    std::string languageDesc;
+    g_LangCodeExpander.Lookup(info.language, languageDesc);
+    fileItem->SetProperty("stream.language", languageDesc);
+
+    fileItem->SetProperty("stream.resolution",
+                          std::to_string(info.width) + "x" + std::to_string(info.height));
+    fileItem->SetProperty("stream.bitrate",
+                          static_cast<int>(std::lrint(static_cast<double>(info.bitrate) / 1000.0)));
+
+    float fps = static_cast<float>(info.fpsRate);
+    if (fps > 0 && info.fpsScale > 0)
+      fps /= info.fpsScale;
+
+    fileItem->SetProperty("stream.fps", CovertFpsToString(fps));
+
+    fileItem->SetProperty("stream.is3d", !info.stereoMode.empty() && info.stereoMode != "mono");
+    fileItem->SetProperty("stream.stereomode", info.stereoMode);
+
+    fileItem->SetProperty("stream.hdrtype", CStreamDetails::HdrTypeToString(info.hdrType));
+
+    fileItem->SetProperty("stream.isdefault",
+                          static_cast<bool>(info.flags & StreamFlags::FLAG_DEFAULT));
+    fileItem->SetProperty("stream.isforced",
+                          static_cast<bool>(info.flags & StreamFlags::FLAG_FORCED));
+    fileItem->SetProperty("stream.ishearingimpaired",
+                          static_cast<bool>(info.flags & StreamFlags::FLAG_HEARING_IMPAIRED));
+    fileItem->SetProperty("stream.isvisualimpaired",
+                          static_cast<bool>(info.flags & StreamFlags::FLAG_VISUAL_IMPAIRED));
+    if (selectedIndex == index)
+      fileItem->Select(true);
+
+    itemsToDisplay.Add(fileItem);
+  }
+
+  if (itemsToDisplay.IsEmpty())
+    itemsToDisplay.Add(MakeFileItemNone());
+
+  const auto selectedItem = OpenSelectDialog(*dialog, 38031, itemsToDisplay);
```

Comment:
I would create the dialog instance inside `OpenSelectDialog`, not here. So, you safe the parameter. You do nothing with `dialog` here except to create the instance. Same for the other places where you create dialogs in this file.